src/platform/mhv/hooks/useDatadogRum.jsx

```diff
@@ -3,7 +3,34 @@ import { datadogRum } from '@datadog/browser-rum';
 
 import { environment } from '@department-of-veterans-affairs/platform-utilities/exports';
 
+const getCookie = name => {
+  const value = `; ${document.cookie}`;
+  const parts = value.split(`; ${name}=`);
+  if (parts.length === 2)
+    return parts
+      .pop()
+      .split(';')
+      .shift();
+  return null;
+};
+
+const checkDdSCookie = () => {
+  const ddSCookie = getCookie('_dd_s');
+  if (!ddSCookie) {
+    return;
+  }
+
+  const cookieParts = ddSCookie.split('&');
+  const idPart = cookieParts.find(part => part.startsWith('id='));
+
+  if (!idPart) {
+    // Cookie _dd_s does not have an id value. nullify the cookie to reinitialize the DD_RUM.
+    document.cookie = '_dd_s=; Path=/; Expires=Thu, 01 Jan 1970 00:00:01 GMT;';
```

Comment:
QUESTION: how does this interact with other cookies? Does this delete all cookies?

Reply:
no, just the `_dd_s`. it is parsed by `getCookie` function

Reply:
The cookie api is wild; I had the same question, and [mdn says](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie#write_a_new_cookie):

> In the code above, newCookie is a string of form key=value, specifying the cookie to set/update. Note that you can only set/update a single cookie at a time using this method.

Reply:
That is wild. Thanks!